CONTRIBUTING.md

```diff
@@ -33,6 +33,31 @@ If you consider opening a PR, here is some documentation to get you started:
 - vis.gl [developer process](https://www.github.com/visgl/tsc/tree/master/developer-process)
 - [deck.gl API design guidelines](/dev-docs/deckgl-api-guidelines.md)
 
+## Testing examples with modified deck.gl source
+
+Each example can be run so that it is built against the deck.gl source code in this repo instead of building against the installed version of deck.gl. This enables using the examples to debug the main deck.gl library source.
+
+To do so use the `yarn start-local` command present in each example's directory. See [webpack.config.local.js](https://github.com/visgl/deck.gl/blob/master/examples/webpack.config.local.js) for details.
+
+### Working with other vis.gl dependencies
+
+Deck.gl has a number of dependencies that fall under vis.gl, and there may be times when it is necessary to make a change in one of these.
+Thus for development it is necessary to checkout a copy of such a dependency and make local changes.
+
+When running an example using `yarn start-local` you can use local version of [luma.gl](https://github.com/visgl/luma.gl/) or [math.gl](https://github.com/uber-web/math.gl) by appending the `--env.local-luma` or `--env.local-math` option.
+
+### Specific module overrides
+
+To get the local build of deck.gl to pick up the local code rather than the modules from npm, there are two important config files:
+
+- [ocular-dev-tools.config.js](https://github.com/visgl/deck.gl/blob/master/ocular-dev-tools.config.js) - See [Ocular documentation for details](https://uber-web.github.io/docs/dev-tools)
+- [examples/webpack.config.local.js](https://github.com/visgl/deck.gl/blob/master/examples/webpack.config.local.js)
+
+For example if you have the `loaders.gl` repo checked out at the same level as deck.gl and want to work with a local copy of the `mvt` module, you would add the following `alias` to the configuration:
```

Comment:
Move this to a comment in the webpack config file?

Reply:
done